AndelaOSP/art-dashboard
# [#161644562] Fix asset details display

## What does this PR do?

Fixes a bug where the asset details page constantly displays one asset.

## Description of Task to be completed?

```gherkin
### Expected
- Asset details of the current asset should be displayed 
 instead of details of the previously assigned asset.
- The user does not have to reload the page to view accurate details.
- The user should see  toast message for `unsuccessful` assign and unassign.  

### Actual
- The details displayed are of the asset previously assigned.
- You have to reload the page so that asset details are displayed correctly
```

## How should this be manually tested?

1. Log in
2. Click the `Assets` link
3. Click on a single asset
4. Assign the asset to yourself
5. Click the back arrow to go back to the asset list or navigate to the next page of the asset list.
6. Click on any other asset to view the details
7. Unassign the currently displayed asset
8. Repeat steps 5 and 6
9. Verify that the data shown is correct

## What are the relevant pivotal tracker stories?

[161644562](https://www.pivotaltracker.com/story/show/161644562)

## Any background context you want to add?

N/A

## Important notes

N/A

## Packages installed

N/A

## Deployment note

N/A

## Related PRs branch | PR (branch_name) | (pr_link)

N/A

## Todos

## Screenshots

When an asset is not assigned and has no specs

<img width="1680" alt="screenshot 2018-11-05 at 17 51 31" src="https://user-images.githubusercontent.com/31339738/48010016-cc2bca80-e12d-11e8-942e-93de0d8da1ec.png">

When an asset is not assigned and has specs

<img width="1680" alt="screenshot 2018-11-05 at 17 51 31" src="https://user-images.githubusercontent.com/31339738/48010091-f9787880-e12d-11e8-9ee4-e47d8d5b2635.png">

When there's an error assigning an asset

<img width="1678" alt="screenshot 2018-11-05 at 17 30 39" src="https://user-images.githubusercontent.com/31339738/48010159-1ca32800-e12e-11e8-960f-d652f0e79954.png">
